src/generic.jl

```diff
@@ -89,16 +109,32 @@ function colwise!(r::AbstractArray, metric::PreMetric, a::AbstractMatrix, b::Abs
     r
 end
 
-function colwise!(r::AbstractArray, metric::SemiMetric, a::AbstractMatrix, b::AbstractVector)
-    colwise!(r, metric, b, a)
-end
+"""
+    colwise(r::AbstractMatrix, metric::PreMetric, a, b)
+
+Compute distances between corresponding elements of the iterable collections
+`a` and `b` according to distance `metric`.
 
+`a` and `b` must have the same number of elements, `r` must be a vector of length
+`length(a) == length(b)`.
+"""
 function colwise(metric::PreMetric, a, b)
     n = get_common_length(a, b)
     r = Vector{result_type(metric, a, b)}(undef, n)
     colwise!(r, metric, a, b)
 end
 
+"""
+    colwise(r::AbstractMatrix, metric::PreMetric,
+              a::AbstractMatrix, b::AbstractMatrix)
+
+Compute distances between each corresponding columns of `a` and `b` according
+to distance `metric`. Exactly one of `a` or `b` can be a vector, in which case
+the distance between that vector and all columns of the other matrix are computed.
+
+`a` and `b` must have the same number of columns if neither of the two is a
+vector. `r` must be a vector of length `maximum(size(a, 2), size(b, 2))`.
```

Comment:
`a` and `b` can't be vectors with this signature. Also `r` isn't used by this method.

Reply:
You're right, not with this signature. But we do allow other signatures as well, I'll list them above.